packages/ckeditor5-typing/src/utils/findattributerange.js

```diff
@@ -4,38 +4,44 @@
  */
 
 /**
- * @module link/findlinkrange
+ * @module typing/utils/findattributerange
  */
 
 /**
- * Returns a range containing the entire link in which the given `position` is placed.
+ * Returns a range containing the entire {@link module:engine/view/attributeelement~AttributeElement#childCount attribute element}
+ * in which the given `position` is placed.
  *
  * It can be used e.g. to get the entire range on which the `linkHref` attribute needs to be changed when having a
  * selection inside a link.
  *
  * @param {module:engine/model/position~Position} position The start position.
- * @param {String} value The `linkHref` attribute value.
+ * @param {String} attributeName The attribute name.
+ * @param {String} value The attribute value.
  * @returns {module:engine/model/range~Range} The link range.
  */
-export default function findLinkRange( position, value, model ) {
-	return model.createRange( _findBound( position, value, true, model ), _findBound( position, value, false, model ) );
+export default function findAttributeRange( position, attributeName, value, model ) {
```

Comment:
I'm thinking of an API of this util. Looking at it's usages:

```javascript
const position = selection.getFirstPosition();
// ...
const range = findAttributeRange( position, 'foo', selection.getAttribute( 'foo' ), model );
```

The above IMO have 1 extra parameter that is not necessary if not 2.

```
findLinkRangeFromSelection( 'foo', selection, model ); // or even
findLinkRangeFromSelection( 'foo', model ); // But the 3 param is kinda better.
```

/cc @Reinmar

Reply:
Shouldn't it be more like this?

```javascript
findAttributeRange( model, positionToStartSearchFrom, attributeName, attributeValue )
```

Potentially, attributeValue is not necessary as it can be read at the initial position.

Reply:
I'd ditch the `attributeValue` - it is always taken from the selection anyway as the position is.

Reply:
> Potentially, attributeValue is not necessary as it can be read at the initial position.

But it may be actually useful as you may be here: `<$text x=1>aa</$text>[]<$text x=2>bb</$text>` and if you could execute this function with `1` or `2` you'd have a better flexibility.

Reply:
> Potentially, attributeValue is not necessary as it can be read at the initial position.

But it may be actually useful as you may be here: `<$text x=1>aa</$text>[]<$text x=2>bb</$text>` and if you could execute this function with `1` or `2` you'd have a better flexibility.

Reply:
BTW, couldn't this method land in one of the model's classes? Maybe a static TreeWalker's method to not mix the layers?

Reply:
Sure but the usage isn't that flexible - the question is do we need a general or specialised util here.

For specialised, the below is enough:

```javascript
findLinkRangeFromSelection( 'foo', selection, model );
```

For a general one, probably as you described (but all our plugins will use it as:

```javascript
const positionToStartSearchFrom = selection.getFirstPostion();
const attributeName = 'foo';
const attributeValue = selection.getAttribute( attributeName );
findAttributeRange( model, positionToStartSearchFrom, attributeName, attributeValue );
```

Reply:
> BTW, couldn't this method land in one of the model's classes? Maybe a static TreeWalker's method to not mix the layers?

It might be a better solution then (if we want a general util - we might re-use it somewhere where we operate on attribue - I'd check: `Highlight`, `RestrictedEditing` and `Mention` from top of my mind. In those we might do similar things but differently.

Reply:
Yep. Highlight: https://github.com/ckeditor/ckeditor5/blob/6612ea68d5f175ce490a86d2092882bbf070fd80/packages/ckeditor5-highlight/src/highlightcommand.js#L68-L73

Reply:
Mention and Restricted editing have different mechanisms.

Reply:
When I was moving it out from `Link` I had a gut feeling, that such util has to be used somewhere else as well, but was not able to find an example ;)

@Reinmar Do we agree 
- on moving such util somewhere, and using it in `-highlight` and here?
- if we agree on ☝️ , then where? `-engine`
- and which implementation? I like to one from `-highlight` for its readability, but the call stack get's longer, and I'm not sure which one is faster.